scripts/models/garageModel.js

```diff
@@ -0,0 +1,74 @@
+(function(module) {
+
+  var garage = {};
+
+  function ParkedCar(data, splitCar, car) {
+    this.make = splitCar[0];
+    this.model = splitCar[1];
+    this.year = splitCar[2];
+    this.id = splitCar[3];
+    this.content = car.replace(/\//g, '_');
+    this.dataContent = car;
+    this.maintenance = data.actionHolder;
+  }
+
+  garage.allParkedCars = [];
+
+  var renderMaintenace = function(carData) {
+    // console.log(carData);
+    var garageTemplateMaintenance = Handlebars.compile($('#maintenance-template').html());
+    // console.log(garageTemplateMaintenance)
+    return garageTemplateMaintenance(carData);
+  };
+  var renderCar = function(carData) {
+    var garageTemplate = Handlebars.compile($('#car-template').html());
+    // console.log(carData);
+    return garageTemplate(carData);
+  };
+
+  garage.getCarMaintenance = function(callback, splitCar, car) {
+    $.get('/maintenance/actionrepository/findbymodelyearid', {modelyearid: splitCar[3]})
+    .then(function(data) {
+      garage.allParkedCars.push(new ParkedCar(data, splitCar, car));
+    }).then(function() {
+      var currentCar = garage.allParkedCars[(garage.allParkedCars.length) - 1];
```

Comment:
Probably more appropriate to instantiate this new car object and hang onto a direct reference to it rather than relying on the last element of the array. What would happen if two objects were added to the garage in quick succession? Possible async issue?